<launch>
    <include file="$(find fetch_gazebo)/launch/pickplace_playground.launch" />

    <include file="$(find fetch_moveit_config)/launch/moveit_rviz.launch" />

    <!--
    <node name="rviz" type="rviz" pkg="rviz" />
    -->
    <!--
    <include file="$(find fetch_arm_control)/launch/arm_interaction_backend.launch" />
    -->

    <!--
    <include file="$(find agile_grasp2)/launch/robot_detect_grasps.launch">
        <arg name="cloud_topic" value="" />
    </include>
    <arg name="cloud_topic" default="/cloud_sized_base" />
    -->

    <node name="detect_grasps" pkg="agile_grasp2" type="detect_grasps" output="screen"
        launch-prefix="gdb --ex run --args">
        <!-- Input Point Cloud -->
        <!-- 0: *.pcd file, 1: sensor_msgs/PointCloud2, 2: agile_grasp2/CloudSized, 3: agile_grasp2/CloudIndexed -->
        <param name="cloud_type" value="1" />
        <param name="use_service" value="false" />
        <param name="cloud_topic" value="/head_camera/depth_registered/points" />
        <param name="plot_mode" value="0" /> <!-- 0: none, 1: PCL, 2: rviz -->
        <param name="only_plot_output" value="false" />
        <param name="save_hypotheses" value="false" />
        <param name="filter_half_grasps" value="false"/>
        <param name="voxelize" value="true"/>
        <param name="antipodal_mode" value="1" />

        <!-- Hand Search -->
        <!--
            <rosparam param="workspace"> [0.57, 0.87, -0.2, 0.15, -0.25, 1] </rosparam>
        -->
        <rosparam param="workspace">[-10, 10, -10, 10, -10, 10] </rosparam>
        <!--
        <rosparam param="camera_pose">[]</rosparam>
        -->
        <!--
        <rosparam param="camera_pose">
        [
            0, 0.445417,  0.895323,  0.215,
            1, 0,         0,        -0.015,
            0, 0.895323, -0.445417,  0.23,
            0, 0,         0,         1
        ]
        </rosparam>
        -->
        <rosparam param="camera_pose">
        [
            0, 0.445417,  0.895323,  0.215,
            1, 0,         0,        -0.015,
            0, 0.895323, -0.445417,  1.0,
            0, 0,         0,         1
        ]
        </rosparam>
        <param name="num_samples" value="20000" />
        <param name="num_threads" value="4" />
        <param name="nn_radius_taubin" value="0.01" />
        <param name="nn_radius_hands" value="0.1" />
        <param name="num_orientations" value="8" />
        <!--
        <param name="predict_antipodal" value="true" />
        -->
        <!-- 0: quadrics, 1: pcl::NormalEstimationOMP -->
        <param name="normal_estimation_method" value="1" />

        <!-- Hand Geometry -->
        <param name="finger_width" value="0.01" />
        <!--
        <param name="hand_outer_diameter" value="0.09" />
        -->
        <param name="hand_outer_diameter" value="0.20" />
        <param name="hand_depth" value="0.06" />
        <param name="hand_height" value="0.02" />
        <param name="init_bite" value="0.01" />

        <!-- Classifier -->
        <param name="images_directory" value="$(env HOME)/.ros/grasp_images/" />
        <param name="model_file" value="$(find agile_grasp2)/caffe/test_1batch2.prototxt" />
        <param name="trained_file" value="$(find agile_grasp2)/caffe/bottles_boxes_cans_5xNeg.caffemodel" />
        <param name="label_file" value="$(find agile_grasp2)/caffe/labels.txt" />
        <param name="min_score_diff" value="300" />
        <param name="batch_size" value="100" />

        <!-- Clustering -->
        <param name="min_inliers" value="5" />

        <!-- Grasp Selection -->
        <param name="num_selected" value="500" />
        <!--
        <param name="gripper_width_range" value="[0.03, 0.08]" />
        -->
        <param name="gripper_width_range" value="[0.03, 0.20]" />
    </node>

    <node name="pick" type="pick.py" pkg="fetch_picking" />

    <!-- Start MoveIt -->
    <include file="$(find fetch_moveit_config)/launch/move_group.launch" >
        <arg name="info" value="true"/><!-- publish grasp markers -->
    </include>
</launch>
